yi-fuzzy-open/src/Yi/Fuzzy.hs

```diff
@@ -76,133 +88,114 @@ fuzzyOpen = fuzzyOpenWithDepth defaultDepth
 
 -- | Fuzzy-opens the directory to the specified depth. The depth needs
 -- to be at least @1@ for it to do anything meaningful.
-fuzzyOpenWithDepth :: Int -> YiM ()
-fuzzyOpenWithDepth d = case () of
-  _ | d <= 0 -> printMsg "You need at least depth of 1 for fuzzyOpenWithDepth"
-    | otherwise -> do
-    fileList <- fmap (fmap FileItem)
-                     (liftBase (getRecursiveContents d "."))
-    bufList <- fmap (fmap (BufferItem . ident . attributes))
-                    (withEditor (gets (M.elems . buffers)))
-    promptRef <- withEditor (spawnMinibufferE "" (const localKeymap))
-    let initialState =
-            FuzzyState (fileList <> V.fromList bufList)
-                       (Just 0)
-                       ""
-    withGivenBuffer promptRef $ do
-        putBufferDyn initialState
-    withEditor (renderE initialState)
+fuzzyOpenWithDepth :: Natural -> YiM ()
+fuzzyOpenWithDepth (fromIntegral->d) = do
```

Comment:
I am also against using viewpatterns here, because `d` is only used once in this function, why not just put the `fromIntegral` function at the place where `d` is used?